AIPrompt docs (#1883)
telerik/blazor-docs: components/aiprompt/events.md

---
title: Events
page_title: AIPrompt - Events

> description: Events in AIPrompt for Blazor. Learn how to handle the various events the component offers.

Reviewer: ```suggestion
description: Explore the events generated by the AIPrompt for Blazor. Learn how to handle these events and implement custom functionality.
```

* [`OnPromptRequest`](#onpromptrequest)
* [`OnCommandExecute`](#oncommandexecute)
* [`OnOutputRate`](#onoutputrate)
* [`PromptTextChanged`](#promptexttchanged)

## OnPromptRequest

The `OnPromptRequest` event fires when the user clicks on the generate button within the Prompt View or retries a prompt from the Output View.

The event handler receives an argument of type [`AIPromptPromptRequestEventArgs`](/blazor-ui/api/Telerik.Blazor.Components.AIPromptPromptRequestEventArgs). See the [example below](#example).

@[template](/_contentTemplates/common/parameters-table-styles.md#table-layout)

| Property | Type | Description |
| --- | --- | --- |
| `Prompt` | `string` | The prompt text of the request. |
| `Output` | `string` | The output of request based on the prompt text. |
| `IsCancelled` | `bool` | Whether the event is cancelled and the built-in action is prevented. |
| `OutputItem` | `AIPromptOutputItem` | The output item. This property will be populated only when the user retries an existing output. See [`AIPromptOutputItem`](/blazor-ui/api/Telerik.Blazor.Components.AIPromptOutputItem). |

## OnCommandExecute

> The `OnCommandExecute` event fires when the user clicks on a command within the Commands View.

Reviewer: ```suggestion
The `OnCommandExecute` event fires when the user clicks on a command within the Commands view.
```

| --- | --- | --- |
| `Command` | `AIPromptCommandDescriptor` | The executed command. |
| `ParentCommand` | `AIPromptCommandDescriptor` | The parent of the executed command. |
| `Output` | `string` | The output based on the executed command. |
| `IsCancelled` | `bool` | Whether the event is cancelled and the built-in action is prevented. |
| `OutputItem` | `AIPromptOutputItem` | The output item. This property will be populated only when the user retries an existing output. See [`AIPromptOutputItem`](/blazor-ui/api/Telerik.Blazor.Components.AIPromptOutputItem). |

## OnOutputRate

The `OnOutputRate` event fires when the user rates an output.

The event handler receives an argument of type [`AIPromptOutputRateEventArgs`](/blazor-ui/api/Telerik.Blazor.Components.AIPromptOutputRateEventArgs). See the [example below](#example).

@[template](/_contentTemplates/common/parameters-table-styles.md#table-layout)

| Property | Type | Description |
| --- | --- | --- |
| `OutputItem` | `AIPromptOutputItem` | Specifies the output item that is being rated. See [`AIPromptOutputItem`](/blazor-ui/api/Telerik.Blazor.Components.AIPromptOutputItem). |

## PromptTextChanged

> The `PromptTextChanged` event fires when the user changes the prompt text. Use the event to update the AIPrompt prompt when the `PromptText` parameter is set with one-way binding, otherwise the user action will be ignored.

Reviewer: ```suggestion
The `PromptTextChanged` event fires when the user changes the prompt text. Use the event to update the AIPrompt's prompt when the `PromptText` parameter is set with one-way binding, otherwise, the user action will be ignored.
```
